modules/grpc/otel/otel-protobuf-parser.cpp

```diff
@@ -110,6 +112,47 @@ _set_value_with_prefix(LogMessage *msg, std::string &key_buffer, size_t key_pref
   log_msg_set_value_by_name_with_type(msg, key_buffer.c_str(), value.c_str(), value.length(), type);
 }
 
+static const std::string &
+_serialize_ArrayValue(const AnyValue &value, LogMessageValueType *type, std::string *buffer)
+{
+  bool is_all_strings = true;
+
+  for (const AnyValue &element : value.array_value().values())
+    {
+      if (element.value_case() == AnyValue::kStringValue)
+        continue;
+
+      is_all_strings = false;
+      break;
+    }
+
+  if (!is_all_strings)
+    {
+      *type = LM_VT_PROTOBUF;
+      value.SerializePartialToString(buffer);
+      return *buffer;
+    }
+
+  ScratchBuffersMarker marker;
+  GString *scratch_buffer = scratch_buffers_alloc_and_mark(&marker);
+  bool first = true;
+
+  for (const AnyValue &element : value.array_value().values())
```

Comment:
Optional: I think we could be optimistic here and optimize for a single traversal and the "all-string" case by starting to format the list while checking for "is_all_string" in the same loop. In case we find a non-string value, we could drop the scratch-buffer and fall back to `SerializePartialToString`.

Reply:
Talked IRL, we keep it this way as we are unsure about the probability of having all strings.